src/game/boe.actions.cpp

```diff
@@ -2003,13 +2007,44 @@ void debug_give_item() {
 	int i = get_num_response(0, univ.scenario.scen_items.size()-1, "Which item?", item_names);
 	bool was_ident = univ.scenario.scen_items[i].ident;
 	univ.scenario.scen_items[i].ident = true;
-	univ.party.give_item(univ.scenario.scen_items[i], true);
+	bool given = false;
+	if(has_feature_flag("debug-give-item", "active-pc")){
+		given = univ.current_pc().give_item(univ.scenario.scen_items[i], true);
+		if(!given)
+			ASB("Debug: can't give to " + univ.current_pc().name);
+	}
+	if(!given)
+		given = univ.party.give_item(univ.scenario.scen_items[i], true);
+	if(!given)
+		ASB("Debug: can't give anyone " + univ.scenario.scen_items[i].full_name);
+
 	univ.scenario.scen_items[i].ident = was_ident;
 	print_buf();
 	put_item_screen(stat_window);
 	put_pc_screen(); // In case the item was food or gold
 }
 
+void debug_overburden() {
+	if(recording){
+		record_action("debug_overburden", "");
+	}
+
+	cPlayer& pc = univ.current_pc();
+	cItem item;
+	item.name = "Debug item";
+	item.weight = 0;
+	item.variety = eItemType::NON_USE_OBJECT;
+	// Give the PC weightless objects that do nothing:
```

Comment:
I have two comments on this feature.

1. This seems like a good place to use the eItemPreset constructor of cItem.
2. It's fine using weightless objects, but I feel like there may also be something to be said for using objects that are ridiculously heavy… though I don't remember if BoE actually gives any penalties for being overburdened in weight.

Reply:
I went with weightless thinking it was necessary so the weight limit wouldn't stop the function from filling every slot. Now I see there's a flag GIVE_ALLOW_OVERLOAD that can do that.

Reply:
Now they're heavy and it's a preset.